### PR TITLE
SASS-7067: Copy the prior data to the session

### PR DESCRIPTION
**Pull request by lukastymo**

### CODE REVIEW

File: app/controllers/pensions/PensionsSummaryController.scala [hunk 1]
```diff
@@ -43,15 +45,22 @@ class PensionsSummaryController @Inject() (implicit
...
-    pensionSessionService.getAndHandle(taxYear, request.user) { (pensionsUserData, prior) =>
-      if (pensionsUserData.exists(!_.pensions.isEmpty)) {
-        Future.successful(Ok(pensionsSummaryView(taxYear, pensionsUserData.map(_.pensions), prior)))
-      } else {
-        /* -- initialise pensions session date either with prior data or an empty model -- */
-        val cya = prior.fold(PensionsCYAModel.emptyModels)(pr => AllPensionsData.generateCyaFromPrior(pr))
-        pensionSessionService.createOrUpdateSessionData(request.user, cya, taxYear, isPriorSubmission = prior.isDefined)(
-          errorHandler.handleError(INTERNAL_SERVER_ERROR))(Ok(pensionsSummaryView(taxYear, Some(cya), prior)))
-      }
+    pensionSessionService.getAndHandle(taxYear, request.user) { (pensionsUserData: Option[PensionsUserData], priorData: Option[AllPensionsData]) =>
```
Comment (harry-ritchie) [inline app/controllers/pensions/PensionsSummaryController.scala:48]: We could rename `pensionsUserData` to `sessionData` to be completely clear?

File: test/models/session/PensionCYAMergedWithPriorDataSpec.scala [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      (Some(sessionNoPaymentIntoPension), Some(fullPensionsModel), PensionCYAMergedWithPriorData(expectedFullSession, true))
+    )
+
+    def assertMergeCases(sessionData: Option[PensionsUserData], priorData: Option[AllPensionsData], expectedResult: PensionCYAMergedWithPriorData) =
+      assert(PensionCYAMergedWithPriorData.mergeSessionAndPriorData(sessionData, priorData) === expectedResult)
+
+    "merge session with prior data" in forAll(cases)(assertMergeCases)
+
+    // let's test if every journey with at least one field filled will stop prior data from being populated
+    // (user is in the middle of the change, don't override their data)
```
Comment (harry-ritchie) [inline test/models/session/PensionCYAMergedWithPriorDataSpec.scala:56]: Do we need this comment?

File: app/controllers/pensions/PensionsSummaryController.scala (hunk 1 above)
Comment (lukastymo) [reply app/controllers/pensions/PensionsSummaryController.scala:48]: done

File: test/models/session/PensionCYAMergedWithPriorDataSpec.scala (hunk 2 above)
Comment (lukastymo) [reply test/models/session/PensionCYAMergedWithPriorDataSpec.scala:56]: I think so, the tests are not immediately obvious what they do - I used antipattern numeric pension1, pension2... to not pollute code with long names.

**Review by harry-ritchie — APPROVED** [review]

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**